src/plugins/fs.h

```diff
@@ -82,6 +82,42 @@ gboolean bd_fs_mount (const gchar *device, const gchar *mountpoint, const gchar
 gchar* bd_fs_get_mountpoint (const gchar *device, GError **error);
 
 gboolean bd_fs_resize (const gchar *device, guint64 new_size, GError **error);
+gboolean bd_fs_repair (const gchar *device, GError **error);
+gboolean bd_fs_check (const gchar *device, GError **error);
+
+typedef enum {
+    BD_FS_OFFLINE_SHRINK = 1 << 1,
+    BD_FS_OFFLINE_GROW = 1 << 2,
+    BD_FS_ONLINE_SHRINK = 1 << 3,
+    BD_FS_ONLINE_GROW = 1 << 4
+} BDFsResizeFlags;
+
+/**
+ * BDFSInfo:
+ * @type: filesystem identifier, must be present
+ * @check_util: required utility for consistency checking, "" if not needed and NULL for no support
+ * @repair_util: required utility for repair, "" if not needed and NULL for no support
+ * @resize_util: required utility for resize, "" if not needed and NULL for no support
+ * @resize_mode: resize availability flags, 0 if no support
+ */
+typedef struct BDFSInfo
+{
+    const gchar *type;
+    const gchar *check_util;
+    const gchar *repair_util;
+    const gchar *resize_util;
+    BDFsResizeFlags resize_mode;
+} BDFSInfo;
+
+typedef enum {
+    BD_FS_RESIZE,
+    BD_FS_REPAIR,
+    BD_FS_CHECK
+} BDFsOpType;
+
```

Comment:
Now I'm confused. Are the above ``struct`` and ``enum`` supposed to be public/part of the API? I don't think they should be.